client/dataset_test.go

```diff
@@ -47,4 +51,16 @@ func TestDatasets(t *testing.T) {
 		assert.NoError(t, err)
 		assert.Equal(t, currentDataset, d)
 	})
+
+	t.Run("Update", func(t *testing.T) {
+		updateDataset := &Dataset{
+			Name:            datasetName,
+			Description:     &datasetDescription,
```

Comment:
I'd like to see one of `Description` or `ExpandJSONDepth` set to a new value to test the update